src/planner/logical_plan/query_node/plan_query_node.cpp

```diff
@@ -18,13 +19,29 @@ unique_ptr<LogicalOperator> LogicalPlanGenerator::VisitQueryNode(BoundQueryNode
 		root = move(distinct);
 	}
 
-	if (node.orders.size() > 0) {
+	if (node.orders.size() > 0 && (node.limit >= 0 || node.offset >= 0)) {
```

Comment:
As Hannes' has mentioned previously, converting from a limit and order by to a top N implementation should be done in the optimizer. It is not always beneficial to use a heap to do this, for example if N is very big the standard ORDER BY should perform better. As such, please leave the code in the planner as generating a limit and order, and then introduce an optimizer step in the Optimizer that looks for a limit with a child order by node and converts them into a top N node.